zppy/templates/coupled_global.py

```diff
@@ -569,25 +608,37 @@ def run(parameters):
 
     xlim = [float(year1), float(year2)]
 
-    num_plots = len(plot_list)
+    num_initial_plots = len(plot_list)
+    num_extra_plots = len(extra_plots)
+    num_total_plots = num_initial_plots + num_extra_plots
     nrows = 4
     ncols = 2
     plots_per_page = nrows * ncols
-    num_pages = math.ceil(num_plots / plots_per_page)
+    num_pages = math.ceil(num_total_plots / plots_per_page)
 
-    i = 0
+    counter_initial_plots = 0
+    counter_extra_plots = 0
     # https://stackoverflow.com/questions/58738992/save-multiple-figures-with-subplots-into-a-pdf-with-multiple-pages
     pdf = matplotlib.backends.backend_pdf.PdfPages(f"{figstr}.pdf")
     for page in range(num_pages):
         fig = plt.figure(1, figsize=[13.5, 16.5])
         for j in range(plots_per_page):
-            if i < num_plots:
+            if counter_initial_plots < num_initial_plots:
                 ax = plt.subplot(nrows, ncols, j + 1)
                 try:
-                    PLOT_DICT[plot_list[i]](ax, xlim, exps)
+                    PLOT_DICT[plot_list[counter_initial_plots]](ax, xlim, exps)
                 except KeyError:
-                    raise KeyError(f"Invalid plot name: {plot_list[i]}")
-                i += 1
+                    raise KeyError(
+                        f"Invalid plot name: {plot_list[counter_initial_plots]}"
+                    )
+                counter_initial_plots += 1
+            elif counter_extra_plots < num_extra_plots:
+                ax = plt.subplot(nrows, ncols, j + 1)
+                try:
+                    plot_generic(ax, xlim, exps, extra_plots[counter_extra_plots])
+                except:
+                    print(f"Invalid var={extra_plots[counter_extra_plots]}")
```

Comment:
Initially, I tried running without adding the variables to `ts` above (which also had issues, see above comment).

That causes the following plots to not generate:
```
Invalid var=CO2
Invalid var=CO2_FFF
Invalid var=CO2_OCN
Invalid var=CO2_LND
Invalid var=TMCO2
Invalid var=TMCO2_FFF
Invalid var=TMCO2_LND
Invalid var=TMCO2_OCN
Invalid var=TOTSOMC
Invalid var=TOTECOSYSC
Invalid var=NEE
Invalid var=TOTVEGC
```